src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -214,7 +214,8 @@ public void OnStartActivity(Activity activity, object payload)
                 activity.SetTag(SemanticConventions.AttributeHttpUrl, GetUri(request));
                 activity.SetTag(SemanticConventions.AttributeHttpFlavor, HttpTagHelper.GetFlavorTagValueFromProtocol(request.Protocol));
 
-                var userAgent = request.Headers["User-Agent"].FirstOrDefault();
+                request.Headers.TryGetValue("User-Agent", out var values);
```

Comment:
Check for the result of `TryGetValue` and only then assign the value to `userAgent`.

Reply:
done